### ISSUE TITLE
Mac: some dangling window pane visible in top left corner

### ISSUE THREAD

**Issue by MSoegtropIMC** [andrejv/wxmaxima#6]
After a fresh installation of the new version (see below) I have some sort of dangling pane in the top left corner of the work book which obscures the area below. This goes away when I enable the History Window in the View Menu. When I disable the history window and restart Maxima, it is there again. This pane also contains the history. Here is a screen shot.

![HistoryWindowBug](https://user-images.githubusercontent.com/7895506/75232269-36e79600-57b7-11ea-987f-1448cbfd6d41.png)

```
-------------------------------------------------------------
wxMaxima version: 18.02.0
Maxima version: 5.43.2
Maxima build date: 2020-01-30 17:11:40
Host type: x86_64-apple-darwin19.2.0
System type: Darwin 19.3.0 X86-64
Lisp implementation type: SBCL
Lisp implementation version: 2.0.1
wxMaxima's idea of the directory layout is:
Icon dir: /Applications/MacPorts/wxMaxima.app/Contents/Resources
User configuration dir: /Users/msoegtrop/
Help dir: /Applications/MacPorts/wxMaxima.app/Contents/Resources/help
Maxima lisp dir: /opt/local/bin/../share/maxima/5.43.2
-------------------------------------------------------------
```

**Comment by gunterkoenigsmann** [comment]
With a bit oft luck this bug has been resolved in the last two years. If it has with a bit of luck the .DMG found at https://github.com/wxMaxima-developers/wxmaxima/releases should work instead.

**Comment by MSoegtropIMC** [comment]
This is the latest version installed by MacPorts. I guess this is not what you want. Should I look into updating the MacPorts file?

**Comment by MSoegtropIMC** [comment]
@gunterkoenigsmann : since MacOS Catalina it is pretty hard to run unsigned apps. There is no one time yes any more as there used to be in previous versions. One has to entirely disable gatekeeper with some sudo commands. Is there any signed recent binary?

**Comment by gunterkoenigsmann** [comment]
Unfortunately I neither own a mac nor a signing key nor do I know if there is a way to automatically let the CI sign the application at the end of the build process. Travis allows to include encrypted keys in the wxMaxima repository that are decrypted during the build. But if I add a new key it tends to forget the encryption key for the old ones => The mac platform would need much research by a dev who owns a mac.

**Comment by MSoegtropIMC** [comment]
Well I mostly working on Macs these days. I think the easiest way forward might be to provide an up to date MacPorts file. Self compile stuff doesn't require a signature and also the MacPorts binary file folders have some flags set in the file system to ignore such checks (as far as I know). I will look into this. Definitely I can install wxMaxima and Maxime with MacPorts without an issue. Maxima is also quite current but wxMaxima is really old.

**Comment by gunterkoenigsmann** [comment]
If you can somehow update the MacPorts wxMaxima file MacPorts comes with that would be great!
If you do so: Could you add 'libomp' to wxMaxima's build dependencies? If wxMaxima is compiles with libomp installed it runs way faster - which is a feature that is relatively new and therefore should not be known to the MacPorts file, still.

The only drawback is: As I cannot test wxMaxima on the mac (other than asking a CI if it compiles) there is no guarantee, though, that wxMaxima will actually work on the mac. But normally with the help of a tester I at least eventually manage to find out why it doesn't work.

**Comment by MSoegtropIMC** [comment]
There is anyway something weird going on. I recently did a major update of OSX which means that all MacPorts stuff needs to be recompiled. Before I had wxMaxima 19.05.7 and after the OS update I have 18.02.0. Possibly some new version of a dependency is said to be untested with the latest OSX and some older version only has a lower bound on the OS version, so that I get an older wxMaxima.

But the dangling window issue reported here is the same on both.

Anyway I will look into the MacPorts files and clean this up. Is there a specific version of wxMaxima you would recommend to make a ports file for?

**Comment by MSoegtropIMC** [comment]
@gunterkoenigsmann : can you do me a favour and check if any of the patches applied to the old port still apply? As far as I can tell only this one patch was applied:

(https://github.com/macports/macports-ports/blob/master/math/wxMaxima/files/patch-src_Dirstructure.cpp.diff)

but maybe you can also have a quick look at the other patches in these folders.

Another question: which wxMaxima version should I put in as dependency? 3.0, 3.2, something else?

And which wxMaxima version should I put in? Right now I have the latest 19.X version in as default and the latest 20.X version as optional. If you think it is a good idea I could also put in the master version as option - might it make easier for people to test master.

**Comment by MSoegtropIMC** [comment]
This is solved in 20.4.0
